src/routes/safe/components/Transactions/TxList/ActionModal.tsx

```diff
@@ -29,7 +27,13 @@ export const ActionModal = (): ReactElement | null => {
 
   switch (selectedAction.actionSelected) {
     case 'cancel':
-      return <RejectTxModal isOpen onClose={onClose} gwTransaction={transaction} />
+      return (
```

Comment:
You could remove this `switch` because they share the same props. You could do something like this:

```ts
const Modal = selectedAction.actionSelected === 'cancel' ? RejectTxModal : ApproveTx

return (
  <Modal
    isOpen
    onClose={onClose}
    transaction={transaction as Overwrite<Transaction, { txDetails: ExpandedTxDetails }>} // but don't cast if possible
)
```

Reply:
Nice suggestion! Even more duplicate code removed 👍